Flakey test in HttpClient RoundRobinConnectionPoolTest

**jetty-10.0.x**

> Running org.eclipse.jetty.http.client.RoundRobinConnectionPoolTest.testMultiplexWithMaxUsage([6] FCGI)
> 2021-12-21 05:23:27.800:INFO :oejs.Server:main: jetty-10.0.8-SNAPSHOT; built: 2021-12-21T04:12:01.659Z; git: f849b8f6a5e1f17a36e273dfa1fb1421e2e97ac9; jvm 11.0.13+8
> 2021-12-21 05:23:27.804:INFO :oejs.AbstractConnector:main: Started ServerConnector@60fe0da9{fcgi/1.0, (fcgi/1.0)}{0.0.0.0:44753}
> 2021-12-21 05:23:27.804:INFO :oejs.Server:main: Started Server@3475f9dc{STARTING}[10.0.8-SNAPSHOT,sto=0] @833175ms
> 2021-12-21 05:23:27.812:INFO :oejs.Server:main: Stopped Server@3475f9dc{STOPPING}[10.0.8-SNAPSHOT,sto=0]
> 2021-12-21 05:23:27.812:INFO :oejs.AbstractConnector:main: Stopped ServerConnector@60fe0da9{fcgi/1.0, (fcgi/1.0)}{0.0.0.0:0}
> Running org.eclipse.jetty.http.client.RoundRobinConnectionPoolTest.testMultiplexWithMaxUsage([7] UNIX_DOMAIN)
> [ERROR] Tests run: 21, Failures: 1, Errors: 0, Skipped: 3, Time elapsed: 6.327 s <<< FAILURE! - in org.eclipse.jetty.http.client.RoundRobinConnectionPoolTest
> [ERROR] org.eclipse.jetty.http.client.RoundRobinConnectionPoolTest.testMultiplexWithMaxUsage(Transport)[5]  Time elapsed: 0.772 s  <<< FAILURE!
> java.lang.AssertionError: 
> [59254, 34711, 59254, 59254, 48870, 34711, 34711, 54265, 48870, 48870, 54265, 54265, 47029, 47029, 54232, 56236, 56236, 56236, 45679, 54232, 54232, 47029, 45679, 45679, 59254, 53975, 59254, 59254, 53975, 40022, 53975, 47914, 47914, 47914, 40022, 40022, 56399, 46173, 56399, 56399, 46173, 48873, 46173, 45695, 48873, 48873, 45695, 45695]
> Expected: a value less than or equal to <15>
>      but: <16> was greater than <15>
> 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
> 	at org.eclipse.jetty.http.client.RoundRobinConnectionPoolTest.testMultiplexWithMaxUsage(RoundRobinConnectionPoolTest.java:261)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

--- a/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/RoundRobinConnectionPoolTest.java
+++ b/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/RoundRobinConnectionPoolTest.java
@@ -249,6 +249,14 @@ public class RoundRobinConnectionPoolTest extends AbstractTest<TransportScenario
         assertTrue(clientLatch.await(count, TimeUnit.SECONDS));
         assertEquals(count, remotePorts.size());
 
+        // Unix Domain does not have ports.
+        if (transport == Transport.UNIX_DOMAIN)
+            return;
+
+        // UDP does not have TIME_WAIT so ports may be reused by different connections.
+        if (transport == Transport.H3)
+            return;
+
         // Maps {remote_port -> number_of_times_port_was_used}.
         Map<Integer, Long> results = remotePorts.stream()
             .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
@@ -257,7 +265,6 @@ public class RoundRobinConnectionPoolTest extends AbstractTest<TransportScenario
         // [p1, p2, p3 | p1, p2, p3 | p4, p4, p5 | p6, p5, p7]
         // Opening p5 and p6 was delayed, so the opening of p7 was triggered
         // to replace p4 while p5 and p6 were busy sending their requests.
-        if (transport != Transport.UNIX_DOMAIN)
-            assertThat(remotePorts.toString(), results.size(), lessThanOrEqualTo(count / maxUsage));
+        assertThat(results.toString(), count / maxUsage, lessThanOrEqualTo(results.size()));
     }
 }

--- a/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/RoundRobinConnectionPoolTest.java
+++ b/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/RoundRobinConnectionPoolTest.java
@@ -258,6 +258,6 @@ public class RoundRobinConnectionPoolTest extends AbstractTest<TransportScenario
         // Opening p5 and p6 was delayed, so the opening of p7 was triggered
         // to replace p4 while p5 and p6 were busy sending their requests.
         if (transport != Transport.UNIX_DOMAIN)
-            assertThat(remotePorts.toString(), count / maxUsage, lessThanOrEqualTo(results.size()));
+            assertThat(remotePorts.toString(), results.size(), lessThanOrEqualTo(count / maxUsage));
     }
 }

